src/price-formatting/format.js

```diff
@@ -155,31 +155,40 @@ function getFractionParts(formatFlags, basePart, deciPipsPart, numberFormatting)
     return { basePart, deciPipsPart };
 }
 
-function getPipDecimals(formatFlags) {
+function getPipDecimals(formatFlags, decimals, actualDecimals) {
+    if (formatFlags.UseExtendedDecimals && getFlagPipDecimals(formatFlags) > 0) {
+        const actualRemaining = Math.max(0, actualDecimals - decimals);
+        return Math.max(getFlagPipDecimals(formatFlags), actualRemaining);
+    }
+
+    return getFlagPipDecimals(formatFlags);
+}
+
+function getFlagPipDecimals(formatFlags) {
     if (formatFlags.AllowTwoDecimalPips) {
         return 2;
     } else if (formatFlags.AllowDecimalPips || formatFlags.DeciPipsFraction) {
         return 1;
     }
-    return null;
+    return 0;
 }
 
 function formatPricePartsDecimals(parts, numberFormatting, value, decimals, formatFlags) {
-
-    let actualDecimals;
-
     if (formatFlags.Percentage && formatFlags.NoRounding) {
         throw new Error('No rounding is not supported on percentage');
     }
 
-    if (formatFlags.NoRounding) {
-        actualDecimals = numberFormatting.getActualDecimals(value);
-        if (actualDecimals <= decimals) {
-            formatFlags.NoRounding = false;
-        }
+    const actualDecimals = numberFormatting.getActualDecimals(value);
+
+    if (formatFlags.UseExtendedDecimals && getFlagPipDecimals(formatFlags) === 0) {

```

Comment:
Shouldn't this part be part of getPipDecimals method? This split is a little bit confusing. Would be good to have one clear getPipDecimals which would return required pipDecimals based on passed arguments.

Reply:
There are two cases - when `Allow[Two]DecimalPips` is specified, and when it's not. We want to add the extra decimals in a different location depending on this.

This here is to handle when we don't have `Allow[Two]DecimalPips`. We need to increase the decimals as much as we can up to 8, avoiding trailing zeroes (but keeping them up to specified decimals), so they end up in `First`. `Pips` always remains as two digits.

The case where we do have `Allow[Two]DecimalPips` is handled in `getPipDecimals()` and the extra digits will go into `DeciPips`.